lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -187,6 +187,12 @@ def generate
           models, db_tables = models_and_tables
           models_by_table_name = {}
           models.each do |m|
+            m.try(:field_specs)&.each do |_name, field_spec|
+              if (pre_migration = field_spec.options.delete(:pre_migration))
+                pre_migration.call(field_spec)
```

Comment:
This is the callback hook that lets the field spec update the `limit:` to 4 if it needs to. Waiting until this late is necessary to traverse to the other side of the `belongs_to`.